### DISCUSSION TITLE
Expose _getOrderColumnsFn so we can override it

### DISCUSSION THREAD

**Discussion by jpresr** [TanStack/table#4279] [Ideas] [answered]
For all the features except ordering, the user is able to customize how data is massaged like with getExpandedRowModel,  getGroupedRowModel, etc. When using ordering and grouping, we don't want the grouped columns to be moved to the start of columnOrder and this is the function that calls orderColumns which does that. We are requesting to expose a getOrderColumnsFn similar to what is done here:

https://github.com/TanStack/table/blob/main/packages/table-core/src/features/Expanding.ts#L156

This will require no updates since by default it will use the built in function, but will let the user override it if they want to.

Thanks!

**Comment by tannerlinsley** [comment, marked as answer]
Does groupedColumnMode: false not work for your use cases?

**Reply by jpresr** [reply]
Thanks!
